llvm-general/src/LLVM/General/Internal/OrcJIT.hs

```diff
@@ -138,6 +138,14 @@ findSymbol (IRCompileLayer cl dl _) symbol exportedSymbolsOnly = flip runAnyCont
     (FFI.findSymbol cl dl symbol' exportedSymbolsOnly') FFI.disposeSymbol
   decodeM symbol
 
+findSymbolIn :: IRCompileLayer -> ModuleSet -> MangledSymbol -> Bool -> IO JITSymbol
+findSymbolIn (IRCompileLayer cl dl _) (ModuleSet moduleSet) symbol exportedSymbolsOnly = flip runAnyContT return $ do
```

Comment:
I don't think the clever infix naming from the C++ api works out well here as a standard function. I think it would be clearer to name it findSymbol or findSymbolInModule.